cmd/edge-proxy/main.go

```diff
@@ -68,6 +70,8 @@ func main() {
 	flag.StringVar(&httpsTunnelAddr, "https-tunnel-listen", "", "Listen address for HTTPS (CONNECT) tunnel server over TLS.  Both tunnels can be served at the same time.")
 	flag.StringVar(&httpsTunnelTLSCert, "https-tunnel-tls-cert", "", "For the HTTPS tunnel, specify file name and path to the TLS certificate /path/file.crt")
 	flag.StringVar(&httpsTunnelTLSKey, "https-tunnel-tls-key", "", "For the HTTPS tunnel, specify file name and path to the TLS key /path/file.key")
+	flag.StringVar(&httpsTunnelUsername, "https-tunnel-username", "", "For the HTTPS tunnel only, require this username for basic authentication")
+	flag.StringVar(&httpsTunnelPassword, "https-tunnel-password", "", "For the HTTPS tunnel only, require this password for basic authentication")
 	flag.Parse()
```

Comment:
It's tough for me to tell if these username/password options are for the server-side or client side.  In order to make these command line options clearer, what do you think about naming all the server options: `http[s]-proxy-server-*` and all of the client options `http[s]-proxy-client-*` ?

...alternatively `connect-proxy-server-*` and `connect-proxy-client-*` ?

Reply:
Does the help text not make it clear?  We decided against using "proxy" because that word is overloaded in edge-proxy.  Connect proxy is clear, but since we can serve both plain and TLS at the same time, that gets quite verbose, e.g. `connect-proxy-over-tls-server-username`

Reply:
That's right, I remember the whole overloaded proxy term discussion.  How about just adding more verbose description?  So instead of
> For the HTTPS tunnel only, require this username for basic authentication

...something like:
> For the HTTPS CONNECT proxy server, require this username for basic authentication

Reply:
Do the username/password options conflict with cert/key options?  What happens if the user  tries to specify  both?

Reply:
There's a possible interaction between these options: the username/password only applies to the HTTPS tunnel.  The HTTPS tunnel requires tls cert and key.  I will add validation to make sure that cert and key is specified when user/pass are specified.

Reply:
Validation added.